lib/puma/launcher.rb

```diff
@@ -205,6 +205,21 @@ def close_binder_listeners
       @binder.close_listeners
     end
 
+    def log_thread_status(events)
```

Comment:
Looks like you'll need to refactor this out a bit so that you have two methods - one that prepares a Ruby object representation of threads and backtraces, then one method that logs that info to events.

Reply:
@nateberkopec Should we always use the object representation, or do you want to preserve the existing format for the SIGINFO and use the object representation only for the status app?

Reply:
I think what matters is that the current logged output remains the same, the status app outputs JSON, and that they both use the same method underneath for obtaining that info.